addons/sms/models/mail_thread.py

```diff
@@ -305,13 +305,14 @@ def _notify_thread_by_sms(self, message, recipients_data, msg_vals=False,
                 value['sanitized'] or original
                 for original, value in sanitized.items()
             ]
+            existing_partners_numbers = {vals_dict['number'] for vals_dict in sms_create_vals}
```

Comment:
This seem to me as a less readable solution, since `sanitized` is a dict so I'd have to check for both original and value in here and `tocreate_numbers` is a list, so I'd need to iterate over it either way to remove existing numbers. How do you see it?